packages/stratimikos/test/test_tpetra_belos_ex.cpp

```diff
@@ -0,0 +1,192 @@
+// @HEADER
+// ***********************************************************************
+//
+//         Stratimikos: Thyra-based strategies for linear solvers
+//                Copyright (2006) Sandia Corporation
+//
+// Under terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+// license for use of this work by or on behalf of the U.S. Government.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Roscoe A. Bartlett (rabartl@sandia.gov)
+//
+// ***********************************************************************
+// @HEADER
+
+#include "Teuchos_ParameterList.hpp"
+#include "Teuchos_VerboseObject.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+#include "Teuchos_XMLParameterListHelpers.hpp"
+
+#include "Tpetra_Core.hpp"
```

Comment:
Do we need to declare a package dependency on Tpetra to use these headers?

Reply:
@cgcgcg, it has a dependency implicitly through the package `ThyraTpetraAdapters`:

https://github.com/trilinos/Trilinos/blob/5301398ff4e35da9c1fb92e0cf83e477e27b28fe/packages/stratimikos/cmake/Dependencies.cmake#L4-L8

It seems impossible for the package `ThyraTpetraAdapters` to not be pulling in Tpetra headers in its header files.  But, in general, yes there should be a listed dependency on `TpetraCore`.

NOTE: A future version of TriBITS will support public and private package dependencies.  (Currently, all package dependencies in TriBITS are public.)